### ISSUE TITLE
The last successfull hit doesn't update the board color

### ISSUE THREAD

**Issue by Sokom141** [Sokom141/BattleShip#32]
Right before winning the game, if the last ShipUnit is hitted the background color of the clicked button doesn't change to the color set to indicate an hit.

**Comment by sstanham** [comment]
Hi again! I can try to tackle this bug.

**Comment by Sokom141** [comment]
Hi @sstanham!
Good, I'm going to assign this issue to you, as always let me know if you need anything 😄

**Comment by sstanham** [comment]
Sorry for the delay, I will be working on this issue throughout the next few days.

**Comment by Sokom141** [comment]
No worries, take the time you need

**Comment by sstanham** [comment]
Hi again, sorry for the delay, I'm working on the bug as we speak. I'm looking through the files and I'm trying to find where the code stops if all the ships are destroyed, or runs again if both players have ships remaining. Would this be in the GUI folder?

**Comment by Sokom141** [comment]
Hi @sstanham, sorry for the late replay but University is getting harder 😭 

You will find the code that need the fix inside the `GameBoard.java` file, this class handles a lot of things about the game (and for this should be broken down in more classes and refactored).
The method affected by the bug should be `handleData` if I'm not mistaken. 
That method in the future will also for sure need to be refactored and extracted in another class for easier access to the protocol, but for now you can modify that.
Let me know if you need anything, have fun 😄

**Comment by sstanham** [comment]
Don't worry @Sokom141, I'm with you there on the University grind 😄

I will take a look at the section you believe is the culprit today, but I also have another question; Are you able to walk me through how to play an entire game on the same device? Or is it only possible via LAN and the internet for now?

**Comment by Sokom141** [comment]
Hey @sstanham !
Actually you can use any of these methods to play a game.

On the same device:
Just run two indipendent processes of the game, the one hosting it should choose a port ( I suggest something higher than ~1000 ) and wait for the client. 
The client should then input the localhost IP (127.0.0.1) and the port previously chosen.
That should be enough to start the game on the same machine.

Over LAN/Internet the flow is pretty much the same, just need for the client to input the right local or remote IP and for the server to open the port / create a rule on the firewall if necessary.

Let me know if you're able to play!

**Comment by sstanham** [comment]
Running the code locally returns a "Connection Closed" error, I'm running with a port of 10000. Any suggestions?

**Comment by Sokom141** [comment]
Just to make sure, are you running the latest development branch?

**Comment by sstanham** [comment]
I believe so, allow me to redownload the branch

**Comment by sstanham** [comment]
Yes, the error occurs on the latest branch. On a side note, for the issue at hand, does the last successful hit not update the color for the winner, the loser, or both players?

**Comment by Sokom141** [comment]
If I'm not mistaken for the winner.

As for the error please make sure in the application settings you have a user avatar, without one the game won't be able to start

**Comment by sstanham** [comment]
Ok, I got it to work! Furthermore, I have identified that for the losing player, they are fine. For the winner, it's the fact that it turns the incorrect color. I'll see what I can do😄

**Comment by Sokom141** [comment]
Good news, thank you!

**Comment by sstanham** [comment]
#39 Just submitted a pull request! How I solved it is in the description, feel free to check it out

Edit: I seem to be encountering an issue linking the pull request to this issue. Could you help me with that?

**Comment by Sokom141** [comment]
Good! I will now review it.
As for the link, on my end, I see the PR fine with the syntax #39
